CMakeLists.txt

```diff
@@ -38,9 +38,13 @@ endif ()
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 # Require standard C++14
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_EXTENSIONS OFF)
+elseif(${CMAKE_CXX_STANDARD} LESS 14)
+    message(FATAL_ERROR "Halide requires C++14 or newer")
```

Comment:
...this is quite amusing, as we officially require only C++11, not C++14, and yet apparently we were (and still are) requiring 14? Anyone know how we ended up here?

Reply:
LLVM started requiring C++14. Must have gotten changed around then.

Reply:
Also: `if (NOT CMAKE_CXX_STANDARD)`.

Reply:
Also: `elseif (CMAKE_CXX_STANDARD LESS 14)`